eng/versioning/external_dependencies.txt

```diff
@@ -0,0 +1,209 @@
+io.dropwizard.metrics:metrics-jvm;4.0.5
+io.reactivex.rxjava2:rxjava;2.2.4
+net.java.dev.jna:jna-platform;5.4.0
+org.apache.commons:commons-collections4;4.2
+org.apache.commons:commons-lang3;3.8.1
+org.apache.commons:commons-text;1.6
+org.apache.maven.plugins:maven-antrun-plugin;1.8
+org.apache.maven.plugins:maven-assembly-plugin;2.2
+org.apache.maven.plugins:maven-checkstyle-plugin;3.1.0
+org.apache.maven.plugins:maven-clean-plugin;3.0.0
+org.apache.maven.plugins:maven-compiler-plugin;3.8.1
+org.apache.maven.plugins:maven-eclipse-plugin;2.8
+org.apache.maven.plugins:maven-failsafe-plugin;2.22.0
+org.apache.maven.plugins:maven-help-plugin;2.1.1
+org.apache.maven.plugins:maven-jar-plugin;3.1.2
+org.apache.maven.plugins:maven-javadoc-plugin;3.1.1
+org.apache.maven.plugins:maven-jxr-plugin;3.0.0
+org.apache.maven.plugins:maven-project-info-reports-plugin;3.0.0
+org.apache.maven.plugins:maven-release-plugin;2.5.3
+org.apache.maven.plugins:maven-resources-plugin;2.4.3
+org.apache.maven.plugins:maven-site-plugin;3.7.1
+org.apache.maven.plugins:maven-source-plugin;3.0.1
+org.apache.maven.plugins:maven-surefire-plugin;3.0.0-M3
+org.apidesign.javadoc:codesnippet-doclet;0.32
+org.codehaus.mojo:build-helper-maven-plugin;3.0.0
+org.codehaus.mojo:exec-maven-plugin;1.2.1
+org.codehaus.mojo:properties-maven-plugin;1.0.0
+org.codehaus.mojo:xml-maven-plugin;1.0
+org.eclipse.jetty:jetty-maven-plugin;9.3.22.v20171030
+org.eclipse.m2e:lifecycle-mapping;1.0.0
+org.jacoco:jacoco-maven-plugin;0.8.4
+org.jacoco:org.jacoco.agent;0.8.4
+org.mockito:mockito-core;3.0.0
+org.nanohttpd:nanohttpd;2.3.1
+
+# External Dependency Exceptions
+# This section is for external dependencies whose versions were different than 
+# what was defined in the parent pom.
+# Format;
+# <some unique identifier, area name preferred>_groupId:artifactId;dependency-version
+
+# This is a unique dependency as it is the only <type>test-jar</type> dependency in the
+# data track. It's also using a SNAPSHOT version which should be disallowed but there is
+# going to be some investigation necessary to find, at the very least, a newer version
+# which is, hopefully, not a SNAPSHOT.
+# sdk\batch\microsoft-azure-batch\pom.xml
+# sdk\keyvault\microsoft-azure-keyvault\pom.xml
+# sdk\keyvault\microsoft-azure-keyvault-extensions\pom.xml
+# sdk\keyvault\microsoft-azure-keyvault-test\pom.xml
+test_jar_com.microsoft.azure:azure-mgmt-resources;1.3.1-SNAPSHOT
+
+# sdk\applicationinsights\microsoft-azure-applicationinsights-query\pom.xml
+applicationinsights_com.microsoft.azure:azure-arm-client-runtime;1.6.3
+applicationinsights_com.microsoft.azure:azure-client-runtime;1.6.3
+applicationinsights_com.microsoft.azure:azure-client-authentication;1.6.3
+
+# everything under sdk\cosmos
+cosmos_org.mockito:mockito-core;1.10.19
+cosmos_com.google.guava:guava;27.0.1-jre
+cosmos_io.dropwizard.metrics:metrics-core;4.1.0
+cosmos_io.dropwizard.metrics:metrics-jvm;4.1.0
+cosmos_io.dropwizard.metrics:metrics-graphite;4.1.0
+
```

Comment:
everything under sdk\cosmos ?

Is it everything that only cosmos SDK uses, and no one else uses (just cosmos set of external dependencies) ?

Reply:
It's not that nobody else uses these, it's that cosmos is using a different version (in these cases you've highlight the version is greater than the common version). Every dependency exception is going to receive an issue. If the version is greater than the common version then it'll be for the java team to look at upgrading the common version. If the version is less than the common version then the library owner will receive it for the purposes to look at upgrading their version to the common version. Ultimately, I'd like to see that entire section vanish.